spotinst/resource_spotinst_ocean_aks_np_virtual_node_group_test.go

```diff
@@ -0,0 +1,627 @@
+
+  // --- strategy -------------------------------------------------------------
+
+  spot_percentage      = 100
+  fallback_to_ondemand = true
+
+  // ---------------------------------------------------------------------------
+
+}
+
+`
+
+// endregion
+
+// region OceanAKSNPVirtualNodeGroup: Headrooms
+func TestAccSpotinstOceanAKSNPVirtualNodeGroup_Headrooms(t *testing.T) {
+	vngResourceName := "test-aks-vng"
+	resourceName := createOceanAKSNPVirtualNodeGroupResource(vngResourceName)
+
+	var virtualNodeGroup azure_np.VirtualNodeGroup
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, "azure") },
+		Providers:    TestAccProviders,
+		CheckDestroy: testOceanAKSNPVirtualNodeGroupDestroy,
+
+		Steps: []resource.TestStep{
+			{
+				Config: createOceanAKSNPVirtualNodeGroupTerraform(&AKSNPVirtualNodeGroupConfigMetadata{vngResourceName: vngResourceName, updateBaselineFields: true}, testBaselineOceanAKSNPVirtualNodeGroupHeadrooms_Create),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOceanAKSNPVirtualNodeGroupExists(&virtualNodeGroup, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.cpu_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.memory_per_unit", "512"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.gpu_per_unit", "0"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.num_of_units", "2"),
+				),
+			},
+			{
+				Config: createOceanAKSNPVirtualNodeGroupTerraform(&AKSNPVirtualNodeGroupConfigMetadata{vngResourceName: vngResourceName, updateBaselineFields: true}, testBaselineOceanAKSNPVirtualNodeGroupHeadrooms_Update),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOceanAKSNPVirtualNodeGroupExists(&virtualNodeGroup, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.cpu_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.memory_per_unit", "512"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.gpu_per_unit", "0"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.0.num_of_units", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.cpu_per_unit", "2048"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.memory_per_unit", "1024"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.gpu_per_unit", "2"),
+					resource.TestCheckResourceAttr(resourceName, "headrooms.1.num_of_units", "4"),
+				),
+			},
+		},
+	})
+}
+
+const testBaselineOceanAKSNPVirtualNodeGroupHeadrooms_Create = `
+resource "` + string(commons.OceanAKSNPVirtualNodeGroupResourceName) + `" "%v" {
+  provider = "%v"  
+
+  name  = "testVng"
+
+  ocean_id = "o-751eaa33"
+
+  availability_zones = [
```

Comment:
availability zones not required here in headrooms

Reply:
removed